libraries/Microsoft.Bot.Builder.AI.QnA/DialogHelper.cs

```diff
@@ -0,0 +1,188 @@
+                    await stepContext.Context.SendActivityAsync(message).ConfigureAwait(false);
+
+                    return new DialogTurnResult(DialogTurnStatus.Waiting);
+                }
+            }
+            else
+            {
+                filteredResponse = new QueryResult[] { response.FirstOrDefault() };
+            }
+
+            return await stepContext.NextAsync(new List<QueryResult>(filteredResponse), cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<DialogTurnResult> CallTrain(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            var trainResponses = stepContext.Result as List<QueryResult>;
+            var currentQuery = stepContext.Values[CurrentQuery] as string;
+
+            var reply = stepContext.Context.Activity.Text;
+
+            if (trainResponses.Count > 1)
+            {
+                var qnaResult = trainResponses.Where(kvp => kvp.Questions[0] == reply).FirstOrDefault();
+
+                if (qnaResult != null)
+                {
+                    stepContext.Values[QnAData] = new List<QueryResult>() { qnaResult };
+
+                    var records = new FeedbackRecord[]
+                    {
+                        new FeedbackRecord
+                        {
+                            UserId = stepContext.Context.Activity.Id,
+                            UserQuestion = currentQuery,
+                            QnaId = qnaResult.Id,
+                        }
+                    };
+
+                    var feedbackRecords = new FeedbackRecords { Records = records };
+
+                    // Call Active Learning Train API
+                    await _services.CallTrainAsync(feedbackRecords).ConfigureAwait(false);
+
+                    return await stepContext.NextAsync(new List<QueryResult>() { qnaResult }, cancellationToken).ConfigureAwait(false);
+                }
+                else if (reply.Equals(CardNoMatchText))
+                {
+                    await stepContext.Context.SendActivityAsync(CardNoMatchResponse, cancellationToken: cancellationToken).ConfigureAwait(false);
+                    return await stepContext.EndDialogAsync().ConfigureAwait(false);
+                }
+                else
+                {
+                    return await stepContext.ReplaceDialogAsync(ActiveLearningDialogName, stepContext.ActiveDialog.State["options"], cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            return await stepContext.NextAsync(stepContext.Result, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<DialogTurnResult> CheckForMultiTurnPrompt(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            if (stepContext.Result is List<QueryResult> response && response.Count > 0)
+            {
+                // -Check if prompt exists 
```

Comment:
When will this be fulfilled? Should we file a work item or will be part of the PR?

Reply:
For this, I am waiting for Tom's master merge.

Reply:
The merge is done. Are we doing this before merging thePR?